packages/spacecat-shared-rum-api-client/src/functions/traffic-acquisition.js

```diff
@@ -47,8 +48,81 @@ function transformFormat(trafficSources) {
   }));
 }
 
-function handler(bundles) {
-  const trafficSources = bundles
+/* c8 ignore start */
+/*
+ * throw-away code for a single customer who customized the experimentation engine
+ * this code will be removed once they start using the default exp engine
+ *
+ * this function fetches experiment manifests, then merges variants data into controls data
+ *
+ * ie:
+ *
+ * if the customer runs for an experiment where variants are as following:
+ *   control: /
+ *   challenger-1: /a1/
+ *   challenger-2: /a2/
+ *
+ * then data for the `/a1/` and `/a2` are counted towards `/`'s data
+ */
+async function mergeBundlesWithSameId(bundles) {
+  if (!bundles[0]?.url.includes('bamboohr.com')) return bundles;
+  const manifestUrls = [
+    ...new Set(bundles.flatMap((bundle) => bundle.events
+      .filter((e) => e.checkpoint === 'experiment')
+      .map((e) => e.source))),
+  ].map((experiment) => fetch(`https://www.bamboohr.com/experiments/${experiment}/manifest.json`));
+
+  const experiments = await Promise.all(manifestUrls);
+  const variants = (await Promise.all(experiments.map((e) => e.json().catch(() => {}))))
+    .filter((json) => json && Object.keys(json).length > 0)
+    .flatMap((json) => json.experiences?.data ?? [])
+    .filter((data) => data.Name === 'Pages');
+
+  const mapping = variants.reduce((acc, cur) => {
+    Object.entries(cur)
+      .filter(([k]) => !['Name', 'Control'].includes(k))
+      .forEach(([, v]) => {
+        acc[new URL(v).pathname] = new URL(cur.Control).pathname;
+      });
+    return acc;
+  }, {});
+
+  const variantPaths = Object.keys(mapping);
+
+  const getControlPath = (url) => {
+    const path = new URL(url).pathname;
+    if (variantPaths.includes(path)) return mapping[path];
+    return path;
+  };
+
+  const byIdAndPath = bundles.reduce((acc, cur) => {
+    const controlPath = getControlPath(cur.url);
+    const key = `${cur.id}-${controlPath}`;
+    if (!acc[key]) acc[key] = [];
+    if (variantPaths.includes(new URL(cur.url).pathname)) {
+      // eslint-disable-next-line no-param-reassign
+      cur.url = new URL(controlPath, cur.url).href;
+    }
+    acc[key].push(cur);
+    return acc;
+  }, {});
+
+  const merged = Object.entries(byIdAndPath).flatMap(([, v]) => {
```

Comment:
@ekremney please check if this handles when the control split is 0 as well

Reply:
`mergeBundlesWithSameId` does not take splits into account. It is used to map variants to control group. See the jsdocs